kotlin/internal/jvm/compile.bzl

```diff
@@ -90,6 +90,9 @@ def _compiler_toolchains(ctx):
 def _compiler_friends(ctx, friends):
     """Creates a struct of friends meta data"""
 
+    if len(friends) > 0 and ctx.attr.testonly == False:
```

Comment:
Ouch - this seems a bit tricky - adding an API (attribute on a rule) for friends, then breaking if you use it, seems like a bad pattern.

Was there a reason you didn't want to extend this across the libraries as well?

Reply:
If you dont' want to tackle the full friends problem, that's fine - but in that case, please just add a duplicate of this to both test rules, or make a common test attributes section. I don't mind duplication here - or rather I prefer it to opening up an API we clobber, more broadly in the rule-set.

Reply:
I opted to preserve the previous behavior under the assumption that there was some reason we didn't support the friends attribute outside of tests. I'm happy to just remove it and allow all targets to have friends.

Reply:
No, there's a broader set of concerns to address if we want to open it to libraries. I'll roll that PR. I've already done it twice internally.

For now, the main issue is just that we don't actually expose those properties on any rule that doesn't use them, so we dont' have this "if ! testonly" thing.  Just duplicate the attribute in the android test, and don't include this check, and we're fine. I'll adjust when I add the other functionality.

Reply:
> Just duplicate the attribute in the android test

Not quite sure what you mean here. Are you wanting to remove the `testonly` check and leave the `friends` attribute in `_common_attr` so that it's available to non `_test` targets?

Reply:
No, I mean restore the attribute where it was (in kt_jvm_test), remove the conditional, and add a duplicate of the attribute on kt_android_local_test, so both tests have it, but nothing else. The present logic would actually let a kt_jvm_library use friends if it was testonly=true, which isn't what we want (at this point, in this PR) and it would make for the attribute being available for code-completion on rules that it will ultimately throw errors on, which is bad ergonomics.

Reply:
Right now with the implementation I have here the `friends` attribute has to be added to the `kt_jvm_library` rule because that's what's handling the underlying sources (via `_kt_android_artifact`). Without that attr being exposed we can't support test friends in `kt_android_local_test`.

Reply:
Oooh. Ok - I see it now.  Crap.  That piece I completely misunderstood. 

Ok, there are ways of hacking this to preserve the API surface, but given that I'm going to update this to support it more broadly, lets' just leave it as it is.